src/mahimahi.rs

```diff
@@ -99,3 +103,219 @@ pub trait MahimahiExt: Mahimahi {
 }
 
 impl<T: Mahimahi + ?Sized> MahimahiExt for T {}
+
+/// Load a mahimahi trace to a [`RepeatedBwPatternConfig`].
+///
+/// The trace is a sequence of timestamps, each timestamp represents an opportunity
+/// of sending a packet at that timestamp.
+///
+/// Each timestamp will be converted into a 12Mbps bandwidth lasting for 1 millisecond,
+/// and then accumulated. For example, if the trace is \[1, 1, 5, 6\], then the bandwidth
+/// pattern will be \[24Mbps for 1ms, 0Mbps for 3ms, 12Mbps for 2ms\]. The generated
+/// [`RepeatedBwPatternConfig`] will default to an infinite loop.
+///
+/// **Warning:** In some cases, this trace may slightly deviate from the behavior of mahimahi.
+///
+/// Returns `Err` string if the mahimahi trace is invalid.
+pub fn load_mahimahi_trace(trace: Vec<u64>) -> Result<RepeatedBwPatternConfig, &'static str> {
```

Comment:
I prefer the function to have an argument to control the repeated times, which is an option. None equals to Some(0). There would be a normal case in which we load two mahimahi traces and play one after the other. So it would be convenient for us can set the repeated times from loading.